packages/webdriver/src/bidi/core.ts

```diff
@@ -54,18 +55,61 @@ export class BidiCore {
             return
         }
 
-        this.#waitForConnected = new Promise<boolean>((resolve) => {
-            this.#ws.on('open', () => {
-                log.info('Connected session to Bidi protocol')
+        log.info(`Connecting to webSocketUrl ${this.#webSocketUrl}`)
+        const parsedUrl = new URL(this.#webSocketUrl)
+        // https://github.com/webdriverio/webdriverio/issues/14039
+        const candidateUrls: string[] = [this.#webSocketUrl]
+        if (!isIP(parsedUrl.hostname)) {
+            const [ips4, ips6] = await Promise.all([
+                dns.promises.resolve4(parsedUrl.hostname),
+                dns.promises.resolve6(parsedUrl.hostname),
+            ])
+            const candidateIps = [...ips4, ...ips6]
+            // If the host resolves to a single IP address
+            // then it does not make sense to try additional candidates
+            // as the web socket DNS resolver would do extactly the same
+            if (candidateIps.length > 1) {
+                const hostnameMapper = (ip: string) => this.#webSocketUrl.replace(parsedUrl.hostname, ip)
+                candidateUrls.push(...candidateIps.map(hostnameMapper))
+            }
+        }
```

Comment:
For testing purposes it makes sense to wrap everything from your comment to here into a function like `getCandidateUrls`

Reply:
done